### DISCUSSION TITLE
Why is X11 windowing code producing XLIB raw window handle while using XCB for talking to X11?

### DISCUSSION THREAD

**Discussion by kvark** [rust-windowing/winit#3681] [Q&A]
See https://github.com/rust-windowing/winit/blob/master/src/platform_impl/linux/x11/window.rs#L1860

As far as I see, all the talking to X11 is done via `xcb_connection` aka XCB. Is there a specific reason we aren't producing `RawWindowHandle::Xcb`?

The context of the question is that I have a different windowing system that *does* produce `rwh::Xcb` handle, and it stubbornly refuses to work with window transparency and Vulkan, somehow. So I'm curious about the thought process in `winit`.

**Comment by notgull** [comment]
The main reason is because Xlib is generally more supported among graphics subsystems due to its staying power. So even though XCB is much safer, we keep around an Xlib connection for this purpose. Among others, but we're trying to reduce the surface area, see #3198

**Comment by kchibisov** [comment]
Yeah, GLX and some other stuff can not work without xlib. We also need xim for now, etc.

There could be an option to use just `xcb` in the future though.
